optimization debian package manager tweaks

Rajpratik71 opened the pull request:
By default, Ubuntu or Debian based "apt" or "apt-get" system installs recommended but not suggested packages .

By passing "--no-install-recommends" option, the user lets apt-get know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

Rajpratik71 commented:
Fresh PR of #339

michaelspedersen commented:
I've been doing a few tests with these changes, and except for the below I have not had any issues:

* In tools/deploy/Dockerfile:4 I get an error when ca-certificates is included. Error disappears when it is removed.

Looking through the changes, there are however still several places where there are duplicate mentions of packages, e.g. ca-certificates (e.g. comparison/baseline_nf_performance-packet/vEdge/VNF_vagrant/base_image/vedge_vm_build.sh:5).

Some files that have multiple apt-gets will also try to install apt-utils and ca-certificates multiple times, which could be improved by only keeping them once in the file.

Rajpratik71 commented:
> I've been doing a few tests with these changes, and except for the below I have not had any issues:
> 
> * In tools/deploy/Dockerfile:4 I get an error when ca-certificates is included. Error disappears when it is removed.
> 
> Looking through the changes, there are however still several places where there are duplicate mentions of packages, e.g. ca-certificates (e.g. comparison/baseline_nf_performance-packet/vEdge/VNF_vagrant/base_image/vedge_vm_build.sh:5).
> 
> Some files that have multiple apt-gets will also try to install apt-utils and ca-certificates multiple times, which could be improved by only keeping them once in the file.

Fixed, removed duplicates entries

michaelspedersen commented on comparison/box-by-box-kvm-docker/install_kvm_prereqs.sh:
```diff
@@ -16,8 +16,8 @@ popd
-apt-get install -y qemu libvirt-bin ebtables dnsmasq
-apt-get install -y libxslt-dev libxml2-dev libvirt-dev zlib1g-dev ruby-dev
+apt-get --no-install-recommends install -y apt-utils ca-certificates qemu libvirt-bin ebtables dnsmasq
+apt-get --no-install-recommends install -y apt-utils ca-certificates libxslt-dev libxml2-dev libvirt-dev zlib1g-dev ruby-dev
```
Double apt-utils and ca-certs

taylor commented on comparison/ansible/roles/multus_cni/files/vf_setup.sh:
```diff
@@ -10,7 +10,7 @@ function logtime() {
 
 if [ ! -e "/usr/bin/lspci" ]; then
-  apt-get install pciutils -y
+  apt-get --no-install-recommends install -y apt-utils ca-certificates pciutils -y
```
Remove duplicate -y argument.

taylor commented on comparison/ansible/roles/retr0h.etcd/.travis.yml:
```diff
@@ -7,7 +7,7 @@ services:
 before_install:
   - sudo apt-get -qq update
-  - sudo apt-get install -o Dpkg::Options::="--force-confold" --force-yes -y docker-engine
+  - sudo apt-get --no-install-recommends install -y apt-utils ca-certificates -o Dpkg::Options::="--force-confold" --force-yes -y docker-engine
```
Remove duplicate -y argument.

taylor commented on comparison/ansible/roles/visualization/templates/Dockerfile.j2:
```diff
@@ -3,5 +3,5 @@ FROM fluent/fluentd:v1.6-debian-1
 RUN ["gem", "install", "fluent-plugin-elasticsearch", "--no-document", "--version", "3.5.2"]
 RUN apt-get update \
-  && apt-get install curl -y
+  && apt-get --no-install-recommends install -y apt-utils ca-certificates curl -y
```
Remove duplicate -y argument.

taylor commented:
@Rajpratik71, please remove the duplicate `-y` options rebase with current `master` branch.

Rajpratik71 commented:
> @Rajpratik71, please remove the duplicate `-y` options rebase with current `master` branch.

done as requested

taylor commented on comparison/ansible/roles/visualization/files/fluentd.dockerfile:
```diff
@@ -3,5 +3,5 @@ FROM fluent/fluentd:v1.6-debian-1
 RUN ["gem", "install", "fluent-plugin-elasticsearch", "--no-document", "--version", "3.5.2"]
 RUN apt-get update \
-  && apt-get install curl -y
+  && apt-get --no-install-recommends install -y apt-utils ca-certificates curl -y
```
@Rajpratik71 please remove -y

taylor commented on comparison/baseline_nf_performance-csit/vEdge/CNF/CSP_multichain_shared/Dockerfile:
```diff
@@ -7,7 +7,7 @@ CMD mkdir ~/sockets
 
 RUN apt-get update -y && \
-    apt-get install --allow-unauthenticated -y \
+    apt-get --no-install-recommends install -y --allow-unauthenticated -y \
```
@Rajpratik71 please remove -y

taylor replied:
@Rajpratik71 please remove -y

taylor commented on comparison/baseline_nf_performance-csit/vEdge/VNF_openstack/base_image/vedge_vm_build.sh:
```diff
@@ -7,18 +7,18 @@ set -o xtrace  # print commands during script execution
 then 
     sudo apt-get update -y
-    sudo apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
+    sudo apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
```
@Rajpratik71 please remove -y

taylor commented on comparison/baseline_nf_performance-csit/vEdge/VNF_vagrant/base_image/vedge_vm_build.sh:
```diff
@@ -2,10 +2,10 @@
 set -o xtrace  # print commands during script execution
 
 sudo apt-get update -y
-sudo apt-get install --allow-unauthenticated -y make gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
```
@Rajpratik71 please remove -y

taylor commented on comparison/baseline_nf_performance-csit/vEdge/VNF_vagrant/base_image/vedge_vm_build.sh:
```diff
@@ -2,10 +2,10 @@
 
 sudo apt-get update -y
-sudo apt-get install --allow-unauthenticated -y make gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
+sudo apt-get --no-install-recommends install -y --allow-unauthenticated -y make gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
```
@Rajpratik71 please remove -y

taylor commented on comparison/baseline_nf_performance-packet/use_case_pocs/ipsec/Dockerfile:
```diff
@@ -9,7 +9,7 @@ CMD mkdir ~/sockets
 
 RUN apt-get update -y && \
-    apt-get install --allow-unauthenticated -y \
+    apt-get --no-install-recommends install -y --allow-unauthenticated -y \
```
@Rajpratik71 please remove -y

taylor commented on comparison/baseline_nf_performance-packet/vEdge/CNF/CSC_multichain_shared/Dockerfile:
```diff
@@ -9,7 +9,7 @@ CMD mkdir ~/sockets
 
 RUN apt-get update -y && \
-    apt-get install --allow-unauthenticated -y \
+    apt-get --no-install-recommends install -y --allow-unauthenticated -y \
```
@Rajpratik71 please remove -y

taylor commented on comparison/baseline_nf_performance-packet/vEdge/CNF/CSP_multichain_shared/Dockerfile:
```diff
@@ -9,7 +9,7 @@ CMD mkdir ~/sockets
 
 RUN apt-get update -y && \
-    apt-get install --allow-unauthenticated -y \
+    apt-get --no-install-recommends install -y --allow-unauthenticated -y \
```
@Rajpratik71 please remove -y

taylor commented on comparison/baseline_nf_performance-packet/vEdge/VNF_openstack/base_image/vedge_vm_build.sh:
```diff
@@ -7,18 +7,18 @@ set -o xtrace  # print commands during script execution
 then 
     sudo apt-get update -y
-    sudo apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
+    sudo apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
```
@Rajpratik71 please remove -y

taylor commented on comparison/baseline_nf_performance-packet/vEdge/VNF_vagrant/base_image/vedge_vm_build.sh:
```diff
@@ -2,10 +2,10 @@
 
 sudo apt-get update -y
-sudo apt-get install --allow-unauthenticated -y make gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates
+sudo apt-get --no-install-recommends install -y --allow-unauthenticated -y make gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates
```
@Rajpratik71 please remove -y

taylor commented on comparison/box-by-box-kvm-docker/Pktgen/bootstrap.sh:
```diff
@@ -11,10 +11,10 @@ set -ex
 
 sudo apt-get update -y
-sudo apt-get install linux-headers-$(uname -r) -y
+sudo apt-get --no-install-recommends install -y linux-headers-$(uname -r) -y
```
@Rajpratik71 please remove -y

taylor commented on comparison/box-by-box-kvm-docker/Pktgen/bootstrap.sh:
```diff
@@ -33,7 +33,7 @@ sudo add-apt-repository \
 sudo apt-get update
 
-sudo apt-get install docker-ce -y
+sudo apt-get --no-install-recommends install -y docker-ce -y
```
@Rajpratik71 please remove -y

michaelspedersen replied:
double -y still there

taylor commented on comparison/box-by-box-kvm-docker/vBNG/Container/cnf_vbng_install.sh:
```diff
@@ -4,7 +4,7 @@
 
 apt-get update -y
-apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
+apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
```
@Rajpratik71 please remove -y

taylor commented on comparison/box-by-box-kvm-docker/vBNG/SRIOV_VM/v_bng_install.sh:
```diff
@@ -15,18 +15,18 @@ sudo cp v_bng.sh /etc/init.d
 
 sudo apt-get update -y
-sudo apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
+sudo apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
```
@Rajpratik71 please remove -y

taylor commented on comparison/box-by-box-kvm-docker/vBNG/SRIOV_VM/vnf_vbng_install.sh:
```diff
@@ -2,18 +2,18 @@
 
 #sudo apt-get update -y
-#sudo apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
+#sudo apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
```
@Rajpratik71 please remove -y

taylor commented on comparison/box-by-box-kvm-docker/vBNG/v_bng_install.sh:
```diff
@@ -15,18 +15,18 @@ sudo cp v_bng.sh /etc/init.d
 
 sudo apt-get update -y
-sudo apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
+sudo apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
```
@Rajpratik71 please remove -y

taylor commented on comparison/box-by-box-kvm-docker/vBNG_Extra/Container/cnf_vbng_install.sh:
```diff
@@ -4,7 +4,7 @@
 
 apt-get update -y
-apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
+apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
```
@Rajpratik71 please remove -y

taylor commented on comparison/box-by-box-kvm-docker/vBNG_Extra/Container/multichain/1st/cnf_vbng_install.sh:
```diff
@@ -5,7 +5,7 @@ cd /vBNG
 
 apt-get update -y
-apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
+apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
```
@Rajpratik71 please remove -y

taylor commented on comparison/box-by-box-kvm-docker/vBNG_Extra/Container/multichain/2nd/cnf_vbng_install.sh:
```diff
@@ -5,7 +5,7 @@ cd /vBNG
 
 apt-get update -y
-apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
+apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
```
@Rajpratik71 please remove -y

taylor commented on comparison/box-by-box-kvm-docker/vBNG_Extra/Container/multichain/nchains/base/cnf_vedge_base_install.sh:
```diff
@@ -5,7 +5,7 @@ cd /vEdge
 
 apt-get update -y
-apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
+apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
```
@Rajpratik71 please remove -y

taylor commented on comparison/box-by-box-kvm-docker/vBNG_Extra/VM_build/vbng_vm_build.sh:
```diff
@@ -2,18 +2,18 @@
 
 sudo apt-get update -y
-sudo apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
+sudo apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
```
@Rajpratik71 please remove -y

taylor commented on comparison/box-by-box-kvm-docker/vDNS/build/v_dns_install.sh:
```diff
@@ -2,11 +2,11 @@
 sudo add-apt-repository -s -y  ppa:openjdk-r/ppa
 sudo apt-get update
-sudo apt-get install --allow-unauthenticated -y wget openjdk-8-jdk bind9 bind9utils bind9-doc apt-transport-https ca-certificates kea-dhcp4-server g++ libcurl4-gnutls-dev libboost-dev kea-dev
+sudo apt-get --no-install-recommends install -y --allow-unauthenticated -y wget openjdk-8-jdk bind9 bind9utils bind9-doc apt-transport-https ca-certificates kea-dhcp4-server g++ libcurl4-gnutls-dev libboost-dev kea-dev
```
@Rajpratik71 please remove -y

michaelspedersen replied:
double -y still there

taylor commented on comparison/box-by-box-kvm-docker/vDNSgen/cnf_vdnsgen_install.sh:
```diff
@@ -9,7 +9,7 @@ chmod +x cnf_dns_test.sh
 
 apt-get update -y
-apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates tcpdump iputils-ping dnsutils vim
+apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates tcpdump iputils-ping dnsutils vim
```
@Rajpratik71 please remove -y

taylor commented on comparison/box-by-box-kvm-docker/vDNSgen/v_packetgen_install.sh:
```diff
@@ -16,7 +16,7 @@ chmod +x dns_test.sh
 
 sudo apt-get update -y
-sudo apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
+sudo apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
```
@Rajpratik71 please remove -y

taylor commented on comparison/kubecon18-chained_nf_test/CNF/CSC_multichain/base/cnf_vedge_base_install.sh:
```diff
@@ -5,7 +5,7 @@ cd /vEdge
 
 apt-get update -y
-apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
+apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
```
@Rajpratik71 please remove -y

taylor commented on comparison/kubecon18-chained_nf_test/CNF/cnf_vedge_install.sh:
```diff
@@ -4,7 +4,7 @@
 
 apt-get update -y
-apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
+apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
```
@Rajpratik71 please remove -y

taylor commented on comparison/kubecon18-chained_nf_test/CNF/multichain/base/cnf_vedge_base_install.sh:
```diff
@@ -5,7 +5,7 @@ cd /vEdge
 
 apt-get update -y
-apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
+apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates vim
```
@Rajpratik71 please remove -y

taylor commented on comparison/kubecon18-chained_nf_test/VNF/base_image/vedge_vm_build.sh:
```diff
@@ -2,18 +2,18 @@
 
 sudo apt-get update -y
-sudo apt-get install --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
+sudo apt-get --no-install-recommends install -y --allow-unauthenticated -y make wget gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates -y
```
@Rajpratik71 please remove -y

taylor commented on comparison/kubecon18-chained_nf_test/VNF_openstack/base_image/vedge_vm_build.sh:
```diff
@@ -2,10 +2,10 @@
 
 sudo apt-get update -y
-sudo apt-get install --allow-unauthenticated -y make gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates
+sudo apt-get --no-install-recommends install -y --allow-unauthenticated -y make gcc libcurl4-openssl-dev python-pip bridge-utils apt-transport-https ca-certificates
```
@Rajpratik71 please remove -y

taylor reviewed (changes requested):
duplicate -y still exist.  please remove

Rajpratik71 commented:
@taylor Extremely sorry for duplicates , time and effort you have to waste in reviewing and commenting for each file . I removed all duplicates and also verified that . I don't think anyone left here

michaelspedersen commented on comparison/ansible/roles/retr0h.etcd/.travis.yml:
```diff
@@ -7,7 +7,7 @@ services:
 before_install:
   - sudo apt-get -qq update
-  - sudo apt-get install -o Dpkg::Options::="--force-confold" --force-yes -y docker-engine
+  - sudo apt-get --no-install-recommends install apt-utils ca-certificates -o Dpkg::Options::="--force-confold" --force-yes docker-engine
```
I am not fully confident in only using `--force-yes`. It might be good to keep the `-y` here as well.

michaelspedersen reviewed (changes requested):
Found a couple small changes still required, otherwise good

michaelspedersen reviewed (approved):
lgtm
